minimesos/src/main/java/com/containersol/minimesos/container/AbstractContainer.java

```diff
@@ -228,6 +232,10 @@ public String getClusterId() {
         return (cluster != null) ? cluster.getClusterId() : null;
     }
 
+    protected String[] createEnvironment() {
+        return envVars.entrySet().stream().map(e -> e.getKey() + "=" + e.getValue()).toArray(String[]::new);
```

Comment:
For one of my previous projects I wrote this
```
    private static final String SINGLE_QUOTE = "\'";
    private static final String DOUBLE_QUOTE = "\"";

    private static final String ESCAPED_SINGLE_QUOTE = "'\"'\"'";
    private static final String ESCAPED_DOUBLE_QUOTE = "\\\"";

    private static final String ESCAPED_SINGLE_QUOTE_REGEX = ESCAPED_SINGLE_QUOTE;
    private static final String ESCAPED_DOUBLE_QUOTE_REGEX = "\\\\\"";

    /**
     * Put quotes around the given String if necessary.
     * <p>
     * Originally copied from {@link org.apache.commons.exec.util.StringUtils#quoteArgument(String)}, as it does not seem to handle arguments with escaped values
     * </p>
     *
     * @param argument the argument to be quoted
     * @return the quoted argument
     * @throws IllegalArgumentException If argument contains both types of quotes
     */
    public static String quoteArgument(final String argument) {

        String cleanedArgument = removeOuterQuotes( argument, true );
        String noEscapesArgument = cleanedArgument.replaceAll( ESCAPED_SINGLE_QUOTE_REGEX, "").replaceAll( ESCAPED_DOUBLE_QUOTE_REGEX, "");

        final StringBuilder buf = new StringBuilder();
        if (noEscapesArgument.contains(DOUBLE_QUOTE)) {
            if (noEscapesArgument.contains(SINGLE_QUOTE)) {
                throw new IllegalArgumentException(
                        "Can't handle single and double quotes in same argument");
            }
            return buf.append(SINGLE_QUOTE).append(cleanedArgument).append(
                    SINGLE_QUOTE).toString();
        } else if (noEscapesArgument.contains(SINGLE_QUOTE)
                || noEscapesArgument.contains(" ")) {
            return buf.append(DOUBLE_QUOTE).append(cleanedArgument).append(
                    DOUBLE_QUOTE).toString();
        } else {
            return cleanedArgument;
        }

    }

    private static String removeOuterQuotes( String argument, boolean trim ) {

        String cleanedArgument = (trim) ? argument.trim() : argument;

        if (cleanedArgument.startsWith(SINGLE_QUOTE) && cleanedArgument.endsWith(SINGLE_QUOTE)) {
            return removeOuterQuotes(cleanedArgument.substring(1, cleanedArgument.length() - 1), false);
        }

        if (cleanedArgument.startsWith(DOUBLE_QUOTE) && cleanedArgument.endsWith(DOUBLE_QUOTE)) {
            return removeOuterQuotes(cleanedArgument.substring(1, cleanedArgument.length() - 1), false);
        }

        return cleanedArgument;

    }
```
although I'm failing to find unit tests for this